### PR TITLE
Translate state and lifecycle

### PR DESCRIPTION
**Pull request by pepjo**

### CODE REVIEW

File: content/docs/state-and-lifecycle.md [hunk 1]
```diff
@@ -300,84 +300,84 @@ ReactDOM.render(
...
     counter: state.counter + props.increment
   };
 });
 ```
 
-### State Updates are Merged {#state-updates-are-merged}
+### Les actualitzacions d'estat es fusionen {#state-updates-are-merged}
 
-When you call `setState()`, React merges the object you provide into the current state.
+Quan invoques a `setState()`, React combina l'objecte que vas proporcionar amb l'estat actual.
```
Comment (ragafus) [inline content/docs/state-and-lifecycle.md]: En cas de triar vós:

```suggestion
Quan invoqueu a `setState()`, React combina l'objecte que proporcioneu amb l'estat actual.
```

En cas de triar tu:

```suggestion
Quan invoques a `setState()`, React combina l'objecte que proporciones amb l'estat actual.
```

File: content/docs/state-and-lifecycle.md [hunk 2]
```diff
@@ -300,84 +300,84 @@ ReactDOM.render(
...
 
-For example, your state may contain several independent variables:
+Per exemple, el teu estat pot contenir diverses variables independents:
```
Comment (ragafus) [inline content/docs/state-and-lifecycle.md]: En cas de preferir vós front a tu:

```suggestion
Per exemple, el vostre estat pot contenir diverses variables independents:
```

File: content/docs/state-and-lifecycle.md [hunk 3]
```diff
@@ -389,7 +389,7 @@ For example, your state may contain several independent variables:
   }
 ```
 
-Then you can update them independently with separate `setState()` calls:
+Després pots actualitzar-les independentment amb invocacions separades a `setState()`:
```
Comment (ragafus) [inline content/docs/state-and-lifecycle.md]: En el cas de preferir vós front a tu:

```suggestion
Després podeu actualitzar-les independentment amb invocacions separades a `setState()`:
```

File: content/docs/state-and-lifecycle.md [hunk 4]
```diff
@@ -407,41 +407,41 @@ Then you can update them independently with separate `setState()` calls:
   }
 ```
 
-The merging is shallow, so `this.setState({comments})` leaves `this.state.posts` intact, but completely replaces `this.state.comments`.
+La fusió és superficial, així que `this.setState({comments})` deixa intacte a `this.state.posts`, però substitueix completament `this.state.comments`.
```
Comment (ragafus) [inline content/docs/state-and-lifecycle.md]:
```suggestion
La fusió és superficial, així que `this.setState({comments})` deixa intacte `this.state.posts`, però substitueix completament `this.state.comments`.
```

File: content/docs/state-and-lifecycle.md [hunk 5]
```diff
@@ -407,41 +407,41 @@ Then you can update them independently with separate `setState()` calls:
...
 
-## The Data Flows Down {#the-data-flows-down}
+## Les dades flueixen cap avall {#the-data-flows-down}
 
-Neither parent nor child components can know if a certain component is stateful or stateless, and they shouldn't care whether it is defined as a function or a class.
+Ni els components pares o fills poden saber si un determinat component té o no té estat i no els hauria d'importar si es defineix com una funció o una classe.
```
Comment (ragafus) [inline content/docs/state-and-lifecycle.md]:
```suggestion
Ni els components pares ni els fills poden saber si un determinat component té o no té estat i no els hauria d'importar si es defineix com una funció o una classe.
```

File: content/docs/state-and-lifecycle.md [hunk 6]
```diff
@@ -407,41 +407,41 @@ Then you can update them independently with separate `setState()` calls:
...
 
-This is why state is often called local or encapsulated. It is not accessible to any component other than the one that owns and sets it.
+Per això és que l'estat sovint se l'anomena local o encapsulat. No és accessible des d'un altre component excepte d'aquell que el posseeix i l'assigna.
```
Comment (ragafus) [inline content/docs/state-and-lifecycle.md]:
```suggestion
És per això que a l'estat sovint se l'anomena local o encapsulat. No és accessible des d'un altre component excepte d'aquell que el posseeix i l'assigna.
```

File: content/docs/state-and-lifecycle.md [hunk 7]
```diff
@@ -407,41 +407,41 @@ Then you can update them independently with separate `setState()` calls:
...
+El component `FormattedDate` rebria `date` en els seus props i no sabria si ha vingut de l'estat de `Clock`, dels props de `Clock`, o si s'ha escrit manualment:
 
 ```js
 function FormattedDate(props) {
   return <h2>It is {props.date.toLocaleTimeString()}.</h2>;
 }
 ```
 
-[**Try it on CodePen**](https://codepen.io/gaearon/pen/zKRqNB?editors=0010)
+[**Prova-ho a codepen**](https://codepen.io/gaearon/pen/zKRqNB?editors=0010)
```
Comment (ragafus) [inline content/docs/state-and-lifecycle.md]: En cas de triar vós front a tu:

```suggestion
[**Proveu-ho a codepen**](https://codepen.io/gaearon/pen/zKRqNB?editors=0010)
```

File: content/docs/state-and-lifecycle.md [hunk 8]
```diff
@@ -407,41 +407,41 @@ Then you can update them independently with separate `setState()` calls:
...
 
-This is commonly called a "top-down" or "unidirectional" data flow. Any state is always owned by some specific component, and any data or UI derived from that state can only affect components "below" them in the tree.
+A això comunament se l'anomena flux de dades "descendent" o "unidireccional"  Qualsevol estat sempre és propietat d'algun component específic, i qualsevol dada o interfície d'usuari derivats d'aquest estat només poden afectar els components "sota" d'ells en l'arbre.
 
-If you imagine a component tree as a waterfall of props, each component's state is like an additional water source that joins it at an arbitrary point but also flows down.
+Si imagines un arbre de components com una cascada de props, l'estat de cada component és com una font d'aigua addicional que se li uneix en un punt arbitrari, però també flueix cap avall.
```
Comment (ragafus) [inline content/docs/state-and-lifecycle.md]: En cas de triar vós front a tu:

```suggestion
Si imagineu un arbre de components com una cascada de props, l'estat de cada component és com una font d'aigua addicional que se li uneix en un punt arbitrari, però també flueix cap avall.
```

File: content/docs/state-and-lifecycle.md [hunk 9]
```diff
@@ -460,8 +460,8 @@ ReactDOM.render(
 );
 ```
 
-[**Try it on CodePen**](https://codepen.io/gaearon/pen/vXdGmd?editors=0010)
+[**Prova-ho a codepen**](https://codepen.io/gaearon/pen/vXdGmd?editors=0010)
```
Comment (ragafus) [inline content/docs/state-and-lifecycle.md]: En cas de triar vós front a tu:

```suggestion
[**Proveu-ho a codepen**](https://codepen.io/gaearon/pen/vXdGmd?editors=0010)
```

File: content/docs/state-and-lifecycle.md [hunk 10]
```diff
@@ -1,16 +1,16 @@
...
+title: Estat i cicle de vida
 permalink: docs/state-and-lifecycle.html
 redirect_from:
   - "docs/interactivity-and-dynamic-uis.html"
 prev: components-and-props.html
 next: handling-events.html
 ---
 
-This page introduces the concept of state and lifecycle in a React component. You can find a [detailed component API reference here](/docs/react-component.html).
+Aquesta pàgina introdueix el concepte d'estat i cicle de vida en un component de React. Pots trobar una [referència detallada de l'API d'un component aquí](/docs/react-component.html).
```
[the remark this answers is not in the document]
Comment (seraco) [reply content/docs/state-and-lifecycle.md]: En la meva opinió, jo només empraria  **"tu"** per a les traduccions en aquesta repo. Així com ho veig, **"vós"** de cada cop s'usa menys. A més, en el contexte de programar crec que quan un usuari llegeix la forma **"tu"** es sent adreçat més directament que quan s'usa **"vós"**. Finalment, alguns cops quan un verb es conjuga amb **"vós"** i el pronom no s'especifica, no està clar si el text es refereix al lector o a una tercera part.

In my opinion, I would just use **"tu"** for the translations in this repo.  The way I see it, **"vós"** is being used more and more rarely. Also, in the context of programming I believe that when a user reads **"tu"** form, they feel they are being addressed more directly than when using **"vós"**. Finally, sometimes when a verb is conjugated with **"vós"** and the pronoun is not specified, it is unclear if the text refers to the reader or a third party.

File: content/docs/state-and-lifecycle.md [hunk 11]
```diff
@@ -68,23 +68,23 @@ ReactDOM.render(
...
+L'estat és similar a les props, però és privat i està completament controlat pel component.
 
-## Converting a Function to a Class {#converting-a-function-to-a-class}
+## Convertir una funció en una classe {#converting-a-function-to-a-class}
 
-You can convert a function component like `Clock` to a class in five steps:
+Es pot convertir un component de funció com `Clock` en una classe en cinc passos:
 
-1. Create an [ES6 class](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Classes), with the same name, that extends `React.Component`.
+1. Crear una [classe ES6](https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Classes) amb el mateix nom que hereti de `React.Component`.
```
[the remark this answers is not in the document]
Comment (seraco) [reply content/docs/state-and-lifecycle.md]: Molt interessant aquesta informació. L'única cosa és que jo conjugaria el verb en la segona persona: *Crea*.

This information is very interesting. The only thing I would say is that I would conjugate the verb in the second person: **"Crea"**.

File: content/docs/state-and-lifecycle.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (ragafus) [reply content/docs/state-and-lifecycle.md]: > Also, in the context of programming I believe that when a user reads _**tu**_ form, they feel they are being addressed more directly than when using _**vós**_. 
>
I agree with that point of view (In some other languages they are having the same kind of discussion. For instance in the German translation https://github.com/reactjs/de.reactjs.org/issues/26)

So, I'm going to open an issue for harmonising the translation when after finishing the first iteration.

[the remark this answers is not in the document]
Comment (seraco) [reply content/docs/state-and-lifecycle.md]: Perfect. Good idea about opening the issue. We can take care of that after we translate the **Core Pages**.

File: content/docs/state-and-lifecycle.md (hunk 11 above)
Comment (seraco) [inline content/docs/state-and-lifecycle.md]:
```suggestion
1. Crea una [classe ES6](https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Classes) amb el mateix nom que hereti de `React.Component`.
```

File: content/docs/state-and-lifecycle.md [hunk 12]
```diff
@@ -68,23 +68,23 @@ ReactDOM.render(
...
 
-2. Add a single empty method to it called `render()`.
+2. Afegir un únic mètode buit anomenat `render()`.
```
Comment (seraco) [inline content/docs/state-and-lifecycle.md]:
```suggestion
2. Afegeix un únic mètode buit anomenat `render()`.
```

File: content/docs/state-and-lifecycle.md [hunk 13]
```diff
@@ -68,23 +68,23 @@ ReactDOM.render(
...
 
-3. Move the body of the function into the `render()` method.
+3. Moure el cos de la funció al mètode `render()`.
```
Comment (seraco) [inline content/docs/state-and-lifecycle.md]:
```suggestion
3. Mou el cos de la funció al mètode `render()`.
```

File: content/docs/state-and-lifecycle.md [hunk 14]
```diff
@@ -99,17 +99,17 @@ class Clock extends React.Component {
...
+El mètode `render` s'invocarà cada vegada que ocorre una actualització; però, sempre que renderitzem `<Clock />` en el mateix node del DOM, es farà servir només una única instància de la classe `Clock`. Això ens permet utilitzar característiques addicionals com l'estat local i els mètodes de cicle de vida.
 
-## Adding Local State to a Class {#adding-local-state-to-a-class}
+## Afegir estat local a una classe {#adding-local-state-to-a-class}
 
-We will move the `date` from props to state in three steps:
+Mourem la data des de les props cap a l'estat en tres passos:
 
-1) Replace `this.props.date` with `this.state.date` in the `render()` method:
+1) Substituir `this.props.date` per `this.state.date` al mètode `render()`:
```
Comment (seraco) [inline content/docs/state-and-lifecycle.md]:
```suggestion
1) Substitueix `this.props.date` per `this.state.date` al mètode `render()`:
```

File: content/docs/state-and-lifecycle.md [hunk 15]
```diff
@@ -124,7 +124,7 @@ class Clock extends React.Component {
 }
 ```
 
-2) Add a [class constructor](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Classes#Constructor) that assigns the initial `this.state`:
+2) Afegir un [constructor de classe](https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Classes#Constructor) que assigni el `this.state` inicial:
```
Comment (seraco) [inline content/docs/state-and-lifecycle.md]:
```suggestion
2) Afegeix un [constructor de classe](https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Classes#Constructor) que assigni el `this.state` inicial:
```

File: content/docs/state-and-lifecycle.md [hunk 16]
```diff
@@ -144,7 +144,7 @@ class Clock extends React.Component {
 }
 ```
 
-Note how we pass `props` to the base constructor:
+Nota com vam passar `props` al constructor base:
```
Comment (seraco) [inline content/docs/state-and-lifecycle.md]:
```suggestion
Nota com passem `props` al constructor base:
```

File: content/docs/state-and-lifecycle.md [hunk 17]
```diff
@@ -153,9 +153,9 @@ Note how we pass `props` to the base constructor:
   }
 ```
 
-Class components should always call the base constructor with `props`.
+Els components de classe sempre han d'invocar al constructor base amb `props`.
 
-3) Remove the `date` prop from the `<Clock />` element:
+3) Eliminar la prop `date` de l'element `<Clock />`:
```
Comment (seraco) [inline content/docs/state-and-lifecycle.md]:
```suggestion
3) Elimina la prop `date` de l'element `<Clock />`:
```

**Pull request merged**